File: client/src/app/data/features/tournaments/dtos/security/menu.dto.ts
export interface MenuResponseDto {
  menuId: number;
  menuName: string;
  menuIcon: string;
  menuOrder: number;
  optionId: number;
  optionName: string;
  optionRoute: string;
  optionIcon: string;
}

export interface GroupedMenu {
  id: number;
  name: string;
  icon: string;
  order: number;
  options: {
    name: string;
    route: string;
    icon: string;
  }[];
}

File: api/PatoCup.Application/DTOs/Security/MenuDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PatoCup.Application.DTOs.Security
{
    public class MenuResponseDto
    {
        public int MenuId { get; set; }
        public string MenuName { get; set; } = string.Empty;
        public string MenuIcon { get; set; } = string.Empty;
        public int MenuOrder { get; set; }

        public int OptionId { get; set; }
        public string OptionName { get; set; } = string.Empty;
        public string OptionRoute { get; set; } = string.Empty;
        public string OptionIcon { get; set; } = string.Empty;
    }
}
